doc/source/how-to-run-flower-on-azure.rst

```diff
@@ -0,0 +1,197 @@
+- **Size**: for this guide, we will use ``Standard_D2s_v3 - 2 vcpus, 8GiB memory``
+
+.. tip::
+
+    For resource group, we can create a new group and assign it to all VMs.
+
+When each VM instance has been created the portal will allow you to download the public
+key (.pem) of each instance. Make sure you save this key in safe place and change its
+permissions to user read only, i.e., run the ``chmod 400 <PATH_TO_PEM_FILE>`` command
+for every .pem file.
+
+Once all three VMs are created then navigate to the overview page where all three VMs
+are listed and open every other VM, and copy its Public IP address. Using the Public IP
+address and the public key (after changing the permissions), login to the instances from
+our local machine by running the following command (by default Azure creates the
+``azureuser``):
+
+.. code-block:: bash
+
+    ssh -i <PATH_TO_PEM_FILE> azureuser@<PUBLIC_IP>
+
+VM Networking
+~~~~~~~~~~~~~
+
+During the execution of the Flower application, the server VM (``SuperLink``) will be
+responsible to orchestrate the execution of the application across the client VMs
+(``SuperNode``). When the SuperLink server starts, by default, it listens to the
+following ports: ``{9092, 9093}``. Port `9092` is used to communicate with the
+Federation clients (``SuperNode``) and port ``9093`` to receive and execute Flower
+applications.
+
+Therefore, to enable this communication we need to allow inbound traffic to the server
+VM instance. To achieve this, we need to navigate to the Networking page of the server
+VM in the Microsoft Azure portal. There, we will click the ``Add inbound port rule``. In
+the new window that pops up we perform the following steps:
+
+- **Source**: we select ``IP Addresses``
+- **Source IP addresses/CIDR ranges**: we add the two Public IPs of our client VMs,
+  separated by comma
+- **Destination**: ``Any``
+- **Service**: ``custom``
+- **Destination port ranges**: ``9092``
+- **Protocol**: ``TCP``
+
+The rest of the fields can be left at their default values.
+
+Finally, we need to also open port 9093 to allow receiving and executing incoming
+application requests. To enable this we just need to repeat the steps above, i.e.,
+create a new inbound rule, where for port range we assign port 9093. If we already know
+the Public IP from which our local machine (e.g., laptop) will be submitting
+applications to the Azure cluster, then we just need to specify the Source IP
+address/CIDR range. However, if we want to keep the port widely open we simply need to
+change source to ``Any``.
+
+To be more precise, if we know the Public IP of our machine, then we need to change the
+following:
+
+- **Source IP addresses/CIDR ranges**: add machine's Public IP
+- **Destination port ranges**: ``9093``
+
+Otherwise, we just need to change the following:
+
+- **Source**: ``Any``
+- **Destination port ranges**: ``9093``
+
+Flower Environment
+------------------
+
+Configuration
```

Comment:
I wouldn't indicate here (or in this doc page in general) how to create an app since there is nothing azure-specific to consider. Instead, we could indicate that each VM should have a python env with flower installed and all dependencies ready (.e.g pytorch, etc). The alternative is to use docker in each VM, with the images containing the relevant dependencies. https://flower.ai/docs/framework/docker/index.html

Reply:
Great suggestion. I will rewrite this section and state that VMs should have already configured the Flower environment and if there is a need to use Docker then point them to the above reference.